src/sentry_utils.c

```diff
@@ -227,7 +226,7 @@ sentry__dsn_new(const char *raw_dsn)
     memset(dsn, 0, sizeof(sentry_dsn_t));
     dsn->refcount = 1;
 
-    dsn->raw = sentry__string_clone(raw_dsn);
+    dsn->raw = sentry__string_clonen_or_null(raw_dsn, raw_dsn_len);
     if (!dsn->raw || !dsn->raw[0] || sentry__url_parse(&url, dsn->raw) != 0) {
         goto exit;
```

Comment:
Most `DSN` fields must be deallocated (because `sentry__url_parse()` also creates a couple of clones to fill the `URL` struct). Since this is a private function behind a public options interface, I prefer to leave the deallocation `sentry__dsn_decref()`. This way, we have two clear points for deallocating all cloned strings: one in the options setter and one in `sentry_options_free()` and won't have to track for double-free, etc.